### ISSUE TITLE
revdep CooRTweet check failures with new melt code

### ISSUE THREAD

**Issue by tdhock** [Rdatatable/data.table#5649]
revdep CooRTweet has the following new issues when checking using data.table from github master,
```
* checking tests ...
 ERROR
Running the tests in 'tests/testthat.R' failed.
Last 13 lines of output:
   3.     +-data.table::melt(...)
   4.     \-data.table::melt.data.table(...)
  -- Error ('test-simulated_data.R:220:5'): larger time window -------------------
  Error in `melt.data.table(x, id.vars = c("object_id", "time_delta"), measure.vars = patterns("^content_id", 
      "^id_user"), value.name = c("content_id", "id_user"))`: When 'measure.vars' is a list, 'value.name' must be a character vector of length =1 or =length(measure.vars).
  Backtrace:
      x
   1. \-CooRTweet (local) sim_test(...) at test-simulated_data.R:220:4
   2.   \-CooRTweet::group_stats(result) at test-simulated_data.R:48:4
   3.     +-data.table::melt(...)
   4.     \-data.table::melt.data.table(...)
  
  [ FAIL 9 | WARN 0 | SKIP 0 | PASS 33 ]
  Error: Test failures
  Execution halted
...

* checking re-building of vignette outputs ... ERROR
Error(s) in re-building vignettes:
  ...
--- re-building 'vignette.Rmd' using rmarkdown

Quitting from lines 103-105 [unnamed-chunk-4] (vignette.Rmd)
Error: processing vignette 'vignette.Rmd' failed with diagnostics:
When 'measure.vars' is a list, 'value.name' must be a character vector of length =1 or =length(measure.vars).
--- failed re-building 'vignette.Rmd'

SUMMARY: processing the following file failed:
  'vignette.Rmd'

Error: Vignette re-building failed.
Execution halted
```
git bisect says the first commit for which this happens is https://github.com/Rdatatable/data.table/pull/4731 

From the error message, it looks like this should be fixed on the user side, but I will investigate further.

**Comment by tdhock** [comment]
the problem is in the definition of `CooRTweet::group_stats`, https://github.com/nicolarighetti/CooRTweet/blob/d25e8d72c0561d40063131cefc4cace51091d78d/R/group_stats.r
```
group_stats <- function(x) {
    variable <- time_delta <- id_user <- NULL
    patterns = function(...) NULL

    x_melted <- data.table::melt(x,
        id.vars = c("object_id", "time_delta"),
        measure.vars = patterns("^content_id", "^id_user"),
        value.name = c("content_id", "id_user")
    )
...
```
using data.table CRAN, `patterns()` in melt is always evaluated as `data.table::patterns()`, but using the new code in data.table master, `patterns()` is taken from the local defintion, which returns NULL, and should be changed, so I will ask the author for an update.

Relevant tests which are already present are:
```
  test(1037.412, melt(x, id.vars=1L, measure.vars=list(2L), value.name=c('a', 'b')),
       error="When 'measure.vars' is a list, 'value.name'")
...
DT.wide = data.table(a2=2, b1=1, b2=2)
measure = function(cols)cols # user-defined function for computing measure.vars, same name as data.table::measure but user-defined version should be used.
test(2183.01, melt(DT.wide, measure.vars=measure()), data.table(variable=factor(c("a2","b1","b2")), value=c(2,1,2)))
```
test 2183.01 says that if the user has defined a local version of measure() function, that should be used instead of data.table::measure(). The same logic is used for patterns() and that is implemented in eval_with_cols in data.table/R/utils.R.

**Comment by tdhock** [comment]
I asked the author to send CRAN a fix, https://github.com/nicolarighetti/CooRTweet/issues/16

**Comment by tdhock** [comment]
This issue is now fixed, as of CooRTweet 1.5.0 on CRAN, 5 Sep 2023.
